### PR TITLE
[CARBONDATA-1730] Support skip.header.line.count option for Hive Table in Carbon

### PR DESCRIPTION
**Pull request by sehriff**
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by sehriff** [comment]
Support `skip.header.line.count` option for Hive Table in Carbon

**Comment by sehriff** [comment]
refering to PR https://github.com/apache/spark/pull/14638/files for hive table scan logic code

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1172/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1749/

File: integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonHiveSessionState.scala [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+import org.apache.spark.sql._
+import org.apache.spark.sql.catalyst.analysis.Analyzer
+import org.apache.spark.sql.execution.SparkPlanner
+import org.apache.spark.sql.execution.datasources._
+
+
+/**
+ * A class that holds all session-specific state in a given [[SparkSession]] backed by Hive.
+ */
+private[hive] class CarbonHiveSessionState(sparkSession: SparkSession)
```
Comment (jackylk) [inline integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonHiveSessionState.scala]: Can you just add the necessary part to support `skip.header.line.count` option without copying the whole class from spark?

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1310/

**Comment by chenliang613** [comment]
@sehriff  please squash all commits to one commit

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/1872/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/1453/

**Pull request closed without merging**